### PR TITLE
Center sign in popovers over the browser window so they are easily visible

### PR DESCRIPTION
**Pull request by nick-funk**
## What does this PR do?

Calculates the popover positioning relative to the browser window so it appears centered nicely over the window when spawned.

## How do I test this PR?

- Go to stream
- Select to sign in or register
- Notice popover is perfectly centered over the browser window horizontally, and just slightly down vertically

### CODE REVIEW

File: src/core/client/stream/common/UserBox/UserBoxContainer.tsx [hunk 1]
```diff
@@ -122,12 +122,21 @@ export class UserBoxContainer extends Component<Props> {
...
+    const popupLeft = winLeft + window.outerWidth / 2 - popupWidth / 2;
+    const popupTop = winTop + 100;
+
     return (
       <>
         <Popup
           href={this.authUrl}
           title="Coral Auth"
-          features="menubar=0,resizable=0,width=350,height=450,top=100,left=500"
+          features={`menubar=0,resizable=0,width=350,height=450,top=${popupTop},left=${popupLeft}}`}
```
Comment (wyattjoh) [inline src/core/client/stream/common/UserBox/UserBoxContainer.tsx]: I'd suggest instead to make `features` accept an object with `Record<string, any>` where by default, the `top` and `left` are set as you've defined above for the `Popover` component.

Comment (nick-funk) [reply src/core/client/stream/common/UserBox/UserBoxContainer.tsx]: Can do!

File: src/core/client/ui/components/Popup/Popup.ts [hunk 2]
```diff
@@ -1,5 +1,13 @@
 import { Component } from "react";
 
+interface WindowFeatures {
+  resizable: number;
+  menuBar: number;
```
Comment (wyattjoh) [inline src/core/client/ui/components/Popup/Popup.ts]: The browser API has it all lowercase:

```suggestion
  menubar: number;
```

**Review by wyattjoh — CHANGES_REQUESTED** [review]

File: src/core/client/ui/components/Popup/Popup.ts (hunk 2 above)
Comment (nick-funk) [reply src/core/client/ui/components/Popup/Popup.ts]: Fixed this up, also defaulted the value like we did previously with new naming.

**Review by wyattjoh — APPROVED** [review]

**Pull request merged**